### PR TITLE
Populate encryptors list for basic admin database objects

### PR DESCRIPTION
**Pull request by corivera**
These encryptors are the list of keys and certificates that are selected when setting the encryption option in the backup database dialog. I also made some changes to the backup service call to return the correct operation completion status and to improve error reporting.

### CODE REVIEW

**Review by barbaravaldez — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs [hunk 1]
```diff
@@ -333,21 +334,23 @@ public void InitializeService(IProtocolEndpoint serviceHost)
...
+                    response.Result = false;
                 }
 
+                await requestContext.SendResult(response);
             }
-            else
+            catch (Exception e)
             {
-                response.Result = false;
+                await requestContext.SendError(e);
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs:348]: Consider logging error here too

File: src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs [hunk 2]
```diff
@@ -333,21 +334,23 @@ public void InitializeService(IProtocolEndpoint serviceHost)
...
+                if (sqlConn != null)
+                {
+                    sqlConn.Dispose();
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs:351]: What about if the task starts up successfully but then an error occurs during execution? Are we guaranteed to have StatusChanged with IsCompleted = true called in that case?

**Review by Charles-Gagnon — APPROVED** [review]

File: src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs (hunk 2 above)
Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs:351]: Looking at the SqlTask class, IsCompleted only gets set to false if the status is InProgress or NotStarted. On other results like Failed or Canceled it gets set to true.

Comment (corivera) [reply src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs:351]: https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/TaskServices/SqlTask.cs#L355

**Pull request merged**